added data collection script

robro-ayush opened the pull request.

Commit:
added data collection script

Commit:
changes acc to changes in robrocloud function argument changes

Commit:
transfomed in ROS Package

Commit:
added subscriber for category

UmangRobro commented on launch/dataCollection.launch:
```diff
@@ -0,0 +1,15 @@
+		<param name="image_type" value="jpeg" />
+		<param name="camera_name" value="cam_4" />
+		<param name="cred_path" value="/home/robro/cred.json" />
+                <param name="category" value="overflow" />
+                <param name="image_size" value="100,100" />
```
Where is the topic name for `category` as discussed on the meet

robro-ayush replied:
done. It was to signify that you can skip some params according to your need. But I guess its better to include all params in committed launch file

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+"""
+@file dataCollection.py .
+
+@author Ayush Saini
```
Please do `class` based version no code will be accepted without class and it's doc's as in robro cloud

robro-ayush replied:
I guess no need for class, it's a simple script.

UmangRobro replied:
see it's needed we implement most of all the `ROS` code in class also we need to standardize the coding style as well as docstrings so for that thing we need all things inside a class. So we can make a more abstract layer on it

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+rospy.init_node('dataCollection', anonymous=True)
+
+proj_name = rospy.get_param('~proj_name')
+ros_topic_name = rospy.get_param('~ros_topic_name')
```
Please give proper naming for `ros_topic_name`.

robro-ayush replied:
done

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+except:
+    crop_q = False
+
+grayscale = rospy.get_param('~grayscale')
+
```
put all this variables at one place

robro-ayush replied:
I did it in order according to the launch file, but okay. Done

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = cred_path
+
+client = storage.Client()
+proj_bucket = client.bucket(proj_name)
```
same here. that why class

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+    print('bucket does not exists, creating bucket..')
+    proj_bucket.storage_class = 'STANDARD'
+    try:
+        client.create_bucket(proj_bucket, location='us-central1')
+    except Exception as e:
```
make location as a variable can be changed in the future. Also, get this through launch file

robro-ayush replied:
Sorry for this, will remember next time.

UmangRobro commented on src/dataCollection.py:
```diff
@@ -0,0 +1,124 @@
+    Args:
+        data ([String])
+    """
+    print('Recieved new Category')
```
pls,create debug mode kinda thing so if enable then only print otherwise not

robro-ayush replied:
Sorry for this too, will remember next time.

Commit:
added category_topic_name

It was fine previously, it was to show how you can skip some params according to your need.

Commit:
Update dataCollection.py

Commit:
added GCP storage class and location param

Commit:
converted script to class

Commit:
arranged functions

UmangRobro committed:
exception added
